api/src/main/java/org/infinispan/api/common/annotations/indexing/_private/Options.java

```diff
@@ -58,4 +59,17 @@ public static org.hibernate.search.engine.backend.types.TermVector termVector(Te
             throw new AssertionFailure("Unexpected value for TermVector: " + termVector);
       }
    }
+
+   public static org.hibernate.search.engine.backend.types.VectorSimilarity vectorSimilarity(VectorSimilarity vectorSimilarity) {
```

Comment:
I think that these adapters should not be in the API module, because they create a dependency (although optional or provided) which should not be there

Reply:
FWIW: https://github.com/infinispan/infinispan/pull/9942#discussion_r855203331
https://hibernate.zulipchat.com/#narrow/stream/132092-hibernate-search-dev/topic/custom.20annotations

Reply:
I opened https://hibernate.atlassian.net/browse/HSEARCH-5035. Thank you